share/ipld/nmt_hasher_integration_test.go

```diff
@@ -0,0 +1,162 @@
+package ipld_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	mrand "math/rand"
+
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	ds "github.com/ipfs/go-datastore"
+	dssync "github.com/ipfs/go-datastore/sync"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-node/share/availability/full"
+	availability_test "github.com/celestiaorg/celestia-node/share/availability/test"
+	"github.com/celestiaorg/celestia-node/share/service"
+)
+
+var _ blockstore.Blockstore = (*MockBlockstore)(nil)
```

Comment:
[non-blocking][suggestion]

There is a lot of template code for the mocks, might be cleaner to have that in a separate file since I expect more tests to be added to this file over time.

Reply:
I moved the mocks and new `TestNode` type to the testing package where their "base types" are defined